pymongo/operations.py

```diff
@@ -32,6 +32,15 @@ def __init__(self, filter=None, doc=None, upsert=None):
         self._doc = doc
         self._upsert = upsert
 
+    def __eq__(self, other):
+        if type(other) == type(self):
+            return (other._filter, other._doc, other._upsert) == \
+                   (self._filter, self._doc, self._upsert)
+        return False
```

Comment:
You're attributing too much sophistication to my code :-), this was my first attempt at implementing `__eq__()` in Python and I simply wasn't aware of `NotImplemented` as a potential return value, so good catch there.

I changed the return value in a new commit - the behavior of `assertNotEqual()` in tests doesn't seem to be affected by this change.